pkg/store/gcworker/gc_worker.go

```diff
@@ -680,6 +680,7 @@ func (w *GCWorker) advanceTxnSafePoint(ctx context.Context, target uint64) (newT
 	if err != nil {
 		return 0, errors.Trace(err)
 	}
+	metrics.SafePointGauge.WithLabelValues("txn").Set(float64(result.NewTxnSafePoint))
```

Comment:
The same problem as I said in another PR:

> And another potential issue is that, according to my experiment, grafana losts the precision of timestamps, which are 64bit integers. It's even worse that the Set function of the gauge in golang accepts fload64 values instead of integers, making the precision losing happens even before being uploaded to prometheus.
As the precision of timestamps are quite important, I'm afraid it's not a good idea to upload the timestamp itself. It's impreciseness may cause misleading in problem investigations. Instead, please consider extracting the physical part from the ts, and upload only the physical. People who need the precise values should search in the logs.